tests/features/protocol-create-action.spec.ts

```diff
@@ -0,0 +1,296 @@
+    });
+
+    beforeEach(async () => {
+      sinon.restore(); // wipe all previous stubs/spies/mocks/fakes
+
+      // clean up before each test rather than after so that a test does not depend on other tests to do the clean up
+      await messageStore.clear();
+      await dataStore.clear();
+      await eventLog.clear();
+    });
+
+    after(async () => {
+      await dwn.close();
+    });
+
+    it('should process "create" rule correctly', async () => {
+      // scenario:
+      // verify requester cannot create without a matching "create" rule
+      // verify role authorized create rule
+      // verify authorized author of ancestor create rule
+      // verify authorized recipient of ancestor create rule
+      // verify anyone can create rule
+      // verify create rule does not grant subsequent write (update)
+
+      const alice = await TestDataGenerator.generateDidKeyPersona();
+      const bob = await TestDataGenerator.generateDidKeyPersona();
+      const carol = await TestDataGenerator.generateDidKeyPersona();
+      const daniel = await TestDataGenerator.generateDidKeyPersona();
+
+      // Alice installs a protocol with "can create" rules
+      const protocolDefinition: ProtocolDefinition = {
+        protocol  : 'foo-bar-baz',
+        published : true,
+        types     : {
+          admin : {},
+          foo   : {},
+          bar   : {},
+          baz   : {}
+        },
+        structure: {
+          admin: {
+            $role: true
+          },
+          foo: {
+            $actions: [
+              {
+                role : 'admin',
+                can  : ProtocolAction.Create
+              },
+            ],
+            bar: {
+              $actions: [
+                {
+                  who : 'author',
+                  of  : 'foo',
+                  can : ProtocolAction.Create
+                },
+                {
+                  who : 'recipient',
+                  of  : 'foo',
+                  can : ProtocolAction.Create
+                }
+              ],
+              baz: {
+                $actions: [
+                  {
+                    who : 'anyone',
+                    can : ProtocolAction.Create
+                  }
+                ],
+              }
+            }
+          }
+        }
+      };
+      const protocolsConfig = await ProtocolsConfigure.create({
+        definition : protocolDefinition,
+        signer     : Jws.createSigner(alice)
+      });
+
+      const protocolsConfigureReply = await dwn.processMessage(alice.did, protocolsConfig.message);
+      expect(protocolsConfigureReply.status.code).to.equal(202);
+
+      // Verify Bob cannot create a record without a matching create rule
+      const bobFooBytes = TestDataGenerator.randomBytes(100);
+      const bobUnauthorizedWrite = await RecordsWrite.create(
+        {
+          signer       : Jws.createSigner(bob),
+          protocol     : protocolDefinition.protocol,
```

Comment:
super nit: Do you mind adding carol as the recipient on this one as well so it more closely matches the successful message?

Reply:
Sure thing, done!